src/app/film-card/film-card.component.ts

```diff
@@ -12,16 +12,14 @@ export class FilmCardComponent implements OnInit {
   constructor(private filmCardService: FilmCardService) { }
 
   ngOnInit() {
-
     this.filmName = "Matrix";
     this.getFilms();
   }
 
-  private getFilms() {
+  private getFilms(): Object[] {
     if(!this.filmName) {return;}
     this.filmCardService.getFilms(this.filmName).subscribe(data => {
```

Comment:
необхідно:
зробити перевірку явною `if(this.filmName) { //code }` instead of `if(!this.filmName) {return;}`

опціонально: (на даному етапі наступні пункти необовязкові але надалі вони повинні бути дотримані)
перейменувати `data` в `films`
задати тип який очікується отримати `((films: any[]) => { // code })`
перевірити дані на валідність `if(films && films.length) { // code }`
опрацювати помилки (оператор catch) `.catch((errMsg) => Observable.threw(errMsg))`